name.abuchen.portfolio/src/name/abuchen/portfolio/model/InvestmentPlan.java

```diff
@@ -43,6 +43,9 @@ public enum Type
 
     private long amount;
     private long fees;
+    private long taxes;
+
+    private Type type;
```

Comment:
I totally agree that we need to introduce a ```type``` property and stop deriving the plan type from the properties.

However, to complete this
* we need to update the version of [Client](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/model/Client.java#L32) file because older version of PP will not handle the new properties (taxes and type)
* add a migration to the [ClientFactory](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/model/ClientFactory.java#L838-L841)
* persists the type also in the binary file format ([ProtobufWriter](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/model/ProtobufWriter.java#L724)) - the XML format will include this automatically due to the reflection used by XStream

My proposal is to migrate the investment plans to always have a ```type```. In the current code, older plans do not have a type; only those that have been edited or newly created. This is very error prone when we change code in the future. I propose to have that "guessing logic" in the ClientFactory and from then on we can assume that type is a non-null property.